packages/react/src/components/Tree/Tree.tsx

```diff
@@ -111,10 +120,15 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
             return acc
           }, [])
         : []
-      this.trySetState({ activeIndex })
+      this.setActiveIndexAndTriggerEvent(e, activeIndex)
     },
   }
 
+  setActiveIndexAndTriggerEvent = (e, activeIndex) => {
```

Comment:
nit: just for the sake of preserving semantics in function name, would suggest to rename the function to `try SetActiveIndexAndTriggerEvent` - in that case function's name will clearly advertise that index, in fact, may not be changed (if controlled)